pkg/utils/syncutil/flexible_wait_group.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2023 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package syncutil
+
+import (
+	"sync"
+)
+
+// FlexibleWaitGroup is a flexible version of sync.WaitGroup.
+// It supports adding to the counter after Wait() has been called,
+// which is not allowed in sync.WaitGroup.
+type FlexibleWaitGroup struct {
+	sync.Mutex
+	count int
```

Comment:
Do you think we can provide an interface to acquire count when exiting schedulers 🤔
Maybe it's a redundant thinking :)